drivers/include/net/netdev.h

```diff
@@ -300,75 +307,118 @@ struct netdev {
  */
 typedef struct netdev_driver {
     /**
-     * @brief Send frame
+     * @brief   Start transmission of the given frame and return directly
      *
-     * @pre `(dev != NULL) && (iolist != NULL`
+     * @pre     `(dev != NULL) && (iolist != NULL)`
      *
-     * @param[in] dev       Network device descriptor. Must not be NULL.
-     * @param[in] iolist    IO vector list to send. Elements of this list may
-     *                      have iolist_t::iol_data == NULL or
-     *                      iolist_t::iol_size == 0. However, unless otherwise
-     *                      specified by the device, the *first* element
-     *                      must contain data.
+     * @param[in]   dev     Network device descriptor. Must not be NULL.
+     * @param[in]   iolist  IO vector list to send. Elements of this list may
+     *                      have iolist_t::iol_size == 0 and (in this case only)
+     *                      iolist_t::iol_data == 0.
```

Comment:
Note this change. Previously the API doc read as if a driver would need to expect elements in the `iolist` that have `iolist_t::iol_data == NULL` but `iolist_t::iol_size > 0`. I think this was just a mistake in formulation and not intentionally.